File: src/main/java/com/patres/automation/validation/PointValidation.java
package com.patres.automation.validation;

import javafx.scene.control.Label;
import javafx.scene.control.TextField;
import javafx.scene.control.TextInputControl;

public class PointValidation extends AbstractValidation {

    private final static String invalidMessage = "Point format: (x;y) ";
    private final static String pattern = "\\((\\d+);(\\d+)\\)"; // (12;34)

    public PointValidation(Label validationLabel, TextInputControl validationField) {
        super(validationLabel, validationField);
        validationLabel.setText(invalidMessage);
    }

    public boolean isConditionFulfilled() {
        return getValidationTextField().getText().matches(pattern);
    }
}

File: src/main/java/com/patres/automation/validation/PointValidation.kt
package com.patres.automation.validation

import com.patres.automation.Main
import com.patres.automation.excpetion.PointFormatException
import com.patres.automation.gui.controller.model.TextActionController

class PointValidation(controller: TextActionController) : AbstractValidation(controller) {

    companion object {
        private val invalidMessage = Main.bundle.getString("error.mustBePoint")
        private const val pattern = "\\((\\d+);(\\d+)\\)" // (12;34)
    }

    init {
        validationLabel.text = invalidMessage
    }

    override fun isConditionFulfilled(): Boolean {
        return validationTextField.text.matches(pattern.toRegex())
    }

    override fun throwException() {
        throw PointFormatException(validationTextField.text)
    }

}

